crates/polars-python/src/lazyframe/visitor/expr_nodes.rs

```diff
@@ -552,7 +553,16 @@ pub(crate) fn into_py(py: Python<'_>, expr: &AExpr) -> PyResult<PyObject> {
                     DynLiteralValue::Str(v) => v.into_py_any(py)?,
                     DynLiteralValue::List(_) => todo!(),
                 },
-                LiteralValue::Scalar(sc) => Wrap(sc.as_any_value()).into_py_any(py)?,
+                LiteralValue::Scalar(sc) => {
+                    match sc.as_any_value() {
+                        // AnyValue conversion of duration to python's
+                        // datetime.timedelta drops nanoseconds because
+                        // there is no support for them. See
+                        // https://github.com/python/cpython/issues/59648
+                        AnyValue::Duration(delta, _) => delta.into_py_any(py)?,
```

Comment:
Don't you need the time unit as well?

Reply:
That lives in the dtype which is passed through separately.